### ISSUE TITLE
create board in shadow-dom

### ISSUE THREAD

**Issue by holgerengels** [jsxgraph/jsxgraph#431]
Hi,

I'd like to use jsxgraph instead of geogebra in [KMap](https://kmap.eu) ..

KMap consists of deeply nested WebComponents. So JSXGraph boards have to be created inside shadow dom. This seems to be impossible at the moment, as I have to pass the elementId to initBoard. I get an error stating that the element cannot be found. As a hack, I tried to pass the shadowRoot as the document .. that helps with finding the element, but then it crashes, because ShadowRoot lacks the method createElement. I suppose, using JSXGraph with shadow dom has a lot more implications ..

Actuall this works:

```
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <link rel="stylesheet" type="text/css" href="https://jsxgraph.org/distrib/jsxgraph.css" />
  <script type="text/javascript" src="https://jsxgraph.org/distrib/jsxgraphcore.js"></script>
</head>
<body>

<script type="module">

  class Graph extends HTMLElement {
    constructor () {
      super();
      this.innerHTML = '<div id="box" class="jxvbox" style="width:800px;height:600px;"></div>';
      JXG.JSXGraph.initBoard('box', {boundingbox: [-10, 10, 10, -10], axis:true, renderer: 'svg'});
    }
  }

  customElements.define('graph-element', Graph)
</script>
<graph-element></graph-element>
</body>
</html>
```

This doesn't:

```
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <link rel="stylesheet" type="text/css" href="https://jsxgraph.org/distrib/jsxgraph.css" />
  <script type="text/javascript" src="https://jsxgraph.org/distrib/jsxgraphcore.js"></script>
</head>
<body>

<script type="module">

  class Graph extends HTMLElement {
    constructor () {
      super();
      const shadow = this.attachShadow({ mode: 'open' })
      const el = document.createElement('div')
      el.id = 'box';
      el.class = 'jxgbox';
      el.style.width = '800px';
      el.style.height = '600px';
      shadow.appendChild(el)
      JXG.JSXGraph.initBoard('box', {boundingbox: [-10, 10, 10, -10], axis:true, renderer: 'svg', document: shadow});
    }
  }

  customElements.define('graph-element', Graph)
</script>
<graph-element></graph-element>
</body>
</html>
```

Are there any plans to support shadow dom in the near future?

Thanks, Regards .. Holger

**Comment by holgerengels** [comment]
Well, actually it kind of works with a little hack .. Here I am delegating the method `createElement` and copying the attribute `documentElement` from `shadowRoot` to its `ownerDocument`

```
import 'jsxgraph';
import {html, css, LitElement, PropertyValues} from 'lit';
import {jsxgraphStyles} from "./jsxgraph-css";

export class KmapJsxGraph extends LitElement {
  static styles = [
    jsxgraphStyles,
    css`
    :host {
      display: block;
    }
    #box {
      width: 400px;
      height: 200px;
    }
  `];

  protected firstUpdated(_changedProperties: PropertyValues) {
    // @ts-ignore
    this.shadowRoot.createElement = (...args) => this.shadowRoot.ownerDocument.createElement(args);
    // @ts-ignore
    this.shadowRoot.documentElement = this.shadowRoot.ownerDocument.documentElement;

    const board = JXG.JSXGraph.initBoard('box',
      {
        boundingbox: [-2*Math.PI, 1.5, 2*Math.PI, -1.5],
        axis: true,
        document: this.shadowRoot as unknown as Document,
      });

    board.create('functiongraph', [function(x: number) { return Math.sin(x) }, -2*Math.PI, 2*Math.PI]);
  }

  render() {
    return html`
      <div id="box" class="jxgbox"></div>
    `;
  }
}
```
... though it spits some errors on mouseenter:

```
Uncaught TypeError: Cannot read properties of null (reading 'parentNode')
    at t.Board.updateCSSTransforms (jsxgraphcore.js:67:178163)
    at t.Board.getMousePosition (jsxgraphcore.js:67:122593)
    at t.Board.pointerMoveListener (jsxgraphcore.js:67:143021)
    at HTMLDivElement.o (jsxgraphcore.js:66:20288))
```

**Comment by alfredwassermann** [comment]
Fixed by 6513cbf1b82ba3ed94bfe7d332beb7dadce596a9
